scalyr_agent/scalyr_client.py

```diff
@@ -1540,40 +1542,54 @@ def __set_attributes(self, thread_id, attributes, log_line_attributes=False):
 
         self.__serialization_base = tmp_buffer.getvalue()
 
-    def add_missing_attributes(self, attributes, log_line_attributes=False):
-        """ Adds items attributes to the base_event's attributes if the base_event doesn't
+    def __get_attributes_to_serialize(self):
+        if self.__disable_logfile_addevents_format:
+            result = dict()
+            if self.__parent_event:
+                result = dict(self.__parent_event.__attrs)
+            if self.__attrs:
+                result.update(self.__attrs)
+            return result
+        else:
+            return self.__attrs
```

Comment:
Here you should only return the attributes that are not duplicates from `__parent_event.__attrs`.  So, you'll want to iterate over `__attrs`, checking contents of `__parent_event.__attrs`.